sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilderTests.java

```diff
@@ -279,6 +281,16 @@ public void getNonEncodedBlobName(String originalBlobName) {
         assertTrue(encryptedBlobClient.getBlobUrl().contains(encodedName));
     }
 
+    @ParameterizedTest
+    @ValueSource(longs = { 0, -1, 15, 4L * Constants.GB })
+    public void illegalRegionLength(long regionLength) {
+        // should we have a test for checking whether length > 4MB?
```

Comment:
Yes. I think we should have two more tests added here:

1. A parameterized version where we provide valid region lengths and assert that when we call `getAuthenticatedRegionDataLength`, it matches what we passed in.
2. A test that confirms when we instantiate a new encryption options object and do not set the authenticated region data length the authenticated length value defaults to 4MB